diesel_derives2/src/util.rs

```diff
@@ -45,3 +47,18 @@ fn option_ty_arg(ty: &Type) -> Option<&Type> {
         _ => None,
     }
 }
+
+#[cfg(not(feature = "nightly"))]
+fn root_span(span: Span) -> Span {
+    span
+}
+
+#[cfg(feature = "nightly")]
+/// There's an issue with the resolution of `__diesel_use_everything` if the
+/// derive itself was generated from within a macro. This is a shitty workaround
+/// until we figure out the expected behavior.
+fn root_span(span: Span) -> Span {
+    use std::mem;
+
+    unsafe { mem::transmute(span.unstable().source()) }
```

Comment:
@dtolnay any idea on whether this is a bug or something I actually need to handle? The problem comes from code that looks like this:

```rust
macro_rules! gen_struct {
    () => {
        #[derive(QueryId)]
        struct Foo;
}

gen_struct!();
```

with hygiene turned on, the body of the derive can't see any of the uses that come from `__diesel_use_everything!();`

Reply:
I don't know. :( Serde hasn't hit this because we don't generate macros from macros. File a Rust issue with a minimal repro if you convince yourself it is a bug.